### PR TITLE
Add section about CSS frameworks

### PR DESCRIPTION
**Pull request by JikkuJose**
PR request in related with the issue #49

### CODE REVIEW

File: components/faqs/support-for-css-frameworks.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+render(
+	<div>
+		<Button>Hello from Tachyons</Button>
+	</div>
+);
+`).trim()
+
+const CSSFrameworks = () => (
+  <SectionLayout title="Support for CSS Frameworks">
```
Comment (philpl) [inline components/faqs/support-for-css-frameworks.js]: Can you rephrase this to be a question please? :) e.g. "How can I integrate a CSS framework?"

File: components/faqs/support-for-css-frameworks.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    <p>Class names defined from a referenced CSS file can be used <b>along</b> with the style definitions! This will let styles to be separated from the markup making it very clean & readable.</p>
```
Comment (philpl) [inline components/faqs/support-for-css-frameworks.js]: This is a bit of a clunky intro, since we can be very specific:

> Integrating an existing CSS framework with styled-components is really easy! You can use its existing class-names alongside your components.

Please don't put in stuff like:

> This will let styles to be separated from the markup making it very clean & readable

Since we generally don't want to encourage everyone to use CSS frameworks and this can be somewhat of a fallacy depending on what you mean here, because markup is not related to the following intro.

File: components/faqs/support-for-css-frameworks.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    <p>Consider we are using the <b><a href="http://tachyons.io/">Tachyons</a></b> library. The classes <b>pa4</b>, <b>ba</b>, <b>bg--black-80</b> from the library can be provided along with the style definition so that the markup will be clean without any style related noise.</p>
```
Comment (philpl) [inline components/faqs/support-for-css-frameworks.js]: Can you make this a bit more general? You can just say:

> Consider you have an existing app with some CSS that has the classnames: `.small`, `.medium`, and `.big`.

or something similar of course. Then you also don't have to mention more frameworks below and come straight to the example.

File: components/faqs/support-for-css-frameworks.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    <p>Same can be done with Bootstrap, Foundation or any similar CSS frameworks.</p>
+    <CodeBlock code={sampleTachyonsExample} language="jsx" />
```
Comment (philpl) [inline components/faqs/support-for-css-frameworks.js]: Below here you should link to two important sections:

- The `.attrs` method: https://www.styled-components.com/docs/api#attrs
- Preventing specificity problems with existing CSS: https://www.styled-components.com/docs/advanced#existing-css

File: components/faqs/support-for-css-frameworks.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import React from 'react'
+
+import SectionLayout from '../SectionLayout'
+import CodeBlock from '../CodeBlock'
+
+const sampleTachyonsExample = (`
+const Button = styled.button.attrs({
+  className: 'pa4 ba b--black-80',
```
Comment (philpl) [inline components/faqs/support-for-css-frameworks.js]: Like I'm saying below: Please don't make it tachyons specific and feel free to add 1-3 example classes to our global css (_document.js) to make this example work.

File: components/faqs/support-for-css-frameworks.js (hunk 1 above)
Comment (JikkuJose) [inline components/faqs/support-for-css-frameworks.js]: Sure, will do! I used such a heading because nav bar links weren't questions! And clicking on them weren't letting the anchor tags to work. Anyways, I am changing the heading in the section & nav bar. Perhaps, I will also have the links in nav bar changed?

File: components/faqs/support-for-css-frameworks.js (hunk 2 above)
Comment (JikkuJose) [inline components/faqs/support-for-css-frameworks.js]: Yes, your intro sounds better. Will use it.
> Since we generally don't want to encourage everyone to use CSS frameworks and this can be somewhat of a fallacy depending on what you mean here, because markup is not related to the following intro.

But, am not sure we should disregard acknowledging  other frameworks because people may be reaching here searching for things like "styled components & bootstrap/other frameworks"

File: components/faqs/support-for-css-frameworks.js (hunk 4 above)
Comment (JikkuJose) [inline components/faqs/support-for-css-frameworks.js]: Yes, I wanted to link to `.attrs` but I missed it! Will do!

File: components/faqs/support-for-css-frameworks.js (hunk 5 above)
Comment (JikkuJose) [inline components/faqs/support-for-css-frameworks.js]: Cool, will then add some classes to the global styles.

File: components/faqs/support-for-css-frameworks.js (hunk 2 above)
Comment (mxstbr) [inline components/faqs/support-for-css-frameworks.js]: Let's just not acknowledge it? Just remove the sentence :blush:

Comment (JikkuJose) [inline components/faqs/support-for-css-frameworks.js]: Sure

**Comment by philpl** [comment]
@mxstbr It's looking good, and only the links are missing now.

@JikkuJose thanks for your great work already! :) you can copy some code from here to add the links: https://github.com/styled-components/styled-components-website/blob/master/components/api/primary/styled.js#L72

**Comment by JikkuJose** [comment]
Just submitted a commit with some minor changes; also, was thinking whether to add a README on how to use the repo?

**Comment by philpl** [comment]
@JikkuJose looking almost ready :) it's missing the correct link component. Also can you swap out the "b" tags with the Code component? We generally don't have bold text on the site in any paragraphs.

You mean a read on how to contribute? Yes, we so need to write a separate contributors markdown file and polish up the readme. I'll see what I can do.

**Comment by JikkuJose** [comment]
Done, fixed it!

@philpl yup, a README would be a great start! Also, check the link in the description. It seems to be failing. Also, I think it would be better to have the development started using the default `start` script than the `dev` one.

**Comment by philpl** [comment]
Thanks again for the great section :)
